gpu/features/test/test_normals.cpp

```diff
@@ -371,6 +371,59 @@ TEST(PCL_FeaturesGPU, normals_highlevel_4)
     }
 }
 
+TEST(PCL_FeaturesGPU, normal_estimation_1000_zeroes)
+{
+    // This number is magic, do not set to lower value.
+    // It may affect error reproducibility.
+    const std::size_t N = 1000;
+
+    const float radius_search = 2.0F;
+    const int max_results = 500;
+
+    PointCloud<PointXYZ>::Ptr cloud(new PointCloud<PointXYZ>());
+    cloud->points.assign(N, {0.0, 0.0, 0.0});
+
+    pcl::NormalEstimation<PointXYZ, Normal> ne;
+    ne.setInputCloud(cloud);
+    ne.setSearchMethod(pcl::search::KdTree<PointXYZ>::Ptr (new pcl::search::KdTree<PointXYZ>));
+    ne.setRadiusSearch(radius_search);
+
+    PointCloud<Normal>::Ptr normals(new PointCloud<Normal>());
+    ne.compute(*normals);
+
+    pcl::gpu::NormalEstimation::PointCloud cloud_device;
+    cloud_device.upload(cloud->points);
+
+    pcl::gpu::NormalEstimation ne_device;
+    ne_device.setInputCloud(cloud_device);
+    ne_device.setRadiusSearch(radius_search, max_results);
+
+    pcl::gpu::NormalEstimation::Normals normals_device;
+    ne_device.compute(normals_device);
```

Comment:
Thanks to the patch, we will not crash here anymore.

Reply:
Please put the comment in the code, with a reference to the original issue. This comment is nice, but the code comment will get more eyes once this PR is closed

Reply:
> Please put the comment in the code, with a reference to the original issue

Done